get_png_files_from_drive.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Google, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+from __future__ import print_function
+import pickle
+from os import path, makedirs, walk, listdir
+import io
+import fire
+import sys
+import shutil
+from distutils.dir_util import copy_tree
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+from googleapiclient.http import MediaIoBaseDownload
+
+# If modifying these scopes, delete the file token.pickle.
+SCOPES = ["https://www.googleapis.com/auth/drive"]
+
+
+def main(folder_name="", reporting=False):
+
+    # Remove combined_png dir if it exists
+    if path.exists("./build/combined_png"):
+        shutil.rmtree("./build/combined_png")
+
+    # Create a token.pickle file to store the users session
+    service = get_service()
+
+    if folder_name == "":
+        folder_name = input("Please enter the name of your drive folder: ")
+
+    # Get the folder instance
+    folder_id = get_folder_id(service, folder_name)
+
+    # Create output_dir
+    output_dir = ensure_directory_exists("temp_download_folder")
+
+    # Get the file IDs to download
+    file_list = get_file_list(service, folder_id)
+
+    # Download the files from the drive and put them in the output_dir
+    download_files(service, file_list, output_dir)
+
+    if reporting:
+        # Generate a report on the downloaded files
+        report_on_download(output_dir)
+
+    # Create a combined PNG dir. Files in the png dir are the default
+    merge_png_dirs(output_dir)
+
+
+def get_service():
+    """Autenticate yourself and create a token.pickle to use the google apiclient.
+    The file token.pickle stores the user's access and refresh tokens, and is
+    created automatically when the authorization flow completes for the first
+    time."""
+
+    creds = None
+    if path.isfile("token.pickle"):
```

Comment:
Token should go somewhere outside the git repo or be in gitignore. Otherwise it'll inevitably end up uploaded to GH.

Reply:
It's already in the gitignore ✅